lib/credentials/cognito_identity_credentials.js

```diff
@@ -296,10 +313,14 @@ AWS.CognitoIdentityCredentials = AWS.util.inherit(AWS.Credentials, {
    * @api private
    */
   createClients: function() {
+    var clientConfig = this._clientConfig;
     this.webIdentityCredentials = this.webIdentityCredentials ||
       new AWS.WebIdentityCredentials(this.params);
-    this.cognito = this.cognito ||
-      new CognitoIdentity({params: this.params});
+    if (!this.cognito) {
+      var cognitoConfig = AWS.util.merge({}, clientConfig);
+      cognitoConfig.params = this.params;
+      this.cognito = new CognitoIdentity(cognitoConfig);
+    }
     this.sts = this.sts || new STS();
```

Comment:
I went back and forth on that. Since `STS` uses a global endpoint, it's fine if the region the user passes in isn't us-east-1.

I would probably also have to update WebIdentityCredentials to accept client config, since it calls STS as well.